sphinx_immaterial/theme_result.py

```diff
@@ -0,0 +1,53 @@
+"""A directive designed to reduce example snippets duplication."""
+from docutils.parsers.rst import directives
+from docutils import nodes
+from sphinx.util.docutils import SphinxDirective
+from sphinx.directives.code import container_wrapper
+
+
+class ResultsDirective(SphinxDirective):
```

Comment:
Maybe call this `rst-example` instead since `result` is rather generic.

Reply:
done

Reply:
I also considered adding the directive to the RST domain, but it seemed like overkill.